### DISCUSSION TITLE
How to load Video into a Sprite from a base64 data url?

### DISCUSSION THREAD

**Discussion by Liquidream** [pixijs/pixijs#9424] [Q&A] [answered]
Hi all,
Does anyone have an example of how to load a video into a Sprite from a base64 data url?

I'm using Pixi.js `v7.2.4` and the code below works fine for me when using images (e.g. jpeg/png), 
but when I try it with an mp4, I'm getting an "*uncaught error in promise from ImageResource.ts:174*"
```ts
const base = new BaseTexture(imgData) // eg "data:video/mp4;base64,xxxxxx"
const texture = new Texture(base)
sprite = Sprite.from(texture)
```
Any ideas how to get my code to work for both images AND videos, from base64 data urls?

Thx in advance.

**Comment by SuperSodaSea** [comment, marked as answer]
`await Assets.load(url)` will support both image & video data URLs in PixiJS's next release (7.3.0, not released yet). Here is an example: <https://pixiplayground.com/#/edit/V_Ji0EuuzdokSZOHpp9Dj>.

As for 7.2.4, you can create a texture from a video data URL like this:
```js
const resource = new VideoResource(url);
const texture = Texture.from(resource);
```

**Reply by Liquidream** [reply]
Thanks for the reply, @SuperSodaSea 
Very happy to hear this is all coming in the next version! 😁

In the meantime - I tried your suggestion, but I seem to be having a problem:
https://pixiplayground.com/#/edit/dgKERgNAl5rR-bdBg-Xfn

I suspect it's because I should be using an async or promise somewhere, to wait for texture to resolve before adding - but I'm not sure how.  If you are able to shed some further light - I'd greatly appreciate it.

Thanks again

**Reply by SuperSodaSea** [reply]
Your code seems to work fine, the video data URL just contains a 1x1 white pixel.

**Reply by Liquidream** [reply]
Ah, so it is (1x1 pixel video).
I've since tested it with another sample video and indeed, all plays fine.

Thanks again, @SuperSodaSea 🤓👍

**Comment by Liquidream** [comment]
Just a quick follow-up on this, @SuperSodaSea 

Alas, it appears that the current "workaround" doesn't work on v7.2.4?
(I was chasing my tail as to why it wasn't working in my code-base, then I tried modifying the playground link below to use current release rather than dev, and got the same result...)
https://pixiplayground.com/#/edit/xUDpps4BVV3Tp9YW7WgVM

So, unless I've missed something - looks like I'll just have to be patient and wait for the next ver 😄

**Reply by SuperSodaSea** [reply]
Oh, sorry: `VideoResource` still don't support data URLs in 7.2.4, we fixed it recently in #9389. Maybe we can ask @bigtimebuddy for a 7.2.5 release.

**Reply by bigtimebuddy** [reply]
You don't need to wait for it base64 video support, you can do this:
https://pixiplayground.com/#/edit/ufYbP0vJzkZk7ru2LZFot

### Before

```js
const resource = new PIXI.VideoResource(vidUrl);
```

### After

```js
const element = document.createElement('video');
element.src = vidUrl;
element.preload = "auto";

const resource = new PIXI.VideoResource(element);
```

**Reply by Liquidream** [reply]
Works perfectly - thanks a lot, @bigtimebuddy! 😊👍
